datameta/api/site.py

```diff
@@ -0,0 +1,207 @@
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from pyramid.view import view_config
+from pyramid.request import Request
+from pyramid.httpexceptions import HTTPNoContent, HTTPForbidden, HTTPBadRequest
+
+from dataclasses import dataclass
+from . import DataHolderBase
+from typing import List, Dict
+from ..models import ApplicationSettings
+from .. import resource, security, errors
+from ..resource import resource_by_id
+import datetime
+
+@dataclass
+class AppSettingsResponseElement(DataHolderBase):
+    """Class for Site Request Response from OpenApi"""
+    uuid:       str
+    key:        str
+    value_type: str
+    value:      str
+
+@view_config(
+    route_name="appsettings", 
+    renderer='json', 
+    request_method="GET", 
+    openapi=True
+)
+def get(request: Request) -> List[AppSettingsResponseElement]:
+    """Return all Site Settings"""
+    db = request.dbsession
+
+    # Authenticate the user
+    auth_user = security.revalidate_user(request)
+
+    appsettings = db.query(ApplicationSettings)
+
+    # User has to be site admin to access these settings
+    if not auth_user.site_admin:
+        raise HTTPForbidden()
+
+    settings = []
+
+    for setting in appsettings:
+        if setting.int_value != None:
+            value = setting.int_value
+            value_type = "int"
+        elif setting.str_value != None:
+            value = setting.str_value
+            value_type = "string"
+        elif setting.float_value != None:
+            value = float_value
+            value_type = "float"
+        elif setting.date_value != None:
+            value = date_value
+            value_type = "date"
+        elif setting.time_value != None:
+            value = time_value
+            value_type = "time"
+
+        settings.append(
+            AppSettingsResponseElement(
+                uuid                  =  str(setting.uuid),
+                key                   =  setting.key,
+                value_type            =  value_type,
+                value                 =  value
+            )
+        )
+
+    return settings
+
+@view_config(
+    route_name="appsettings_id",
+    renderer='json',
+    request_method="PUT",
+    openapi=True
+)
+def put(request:Request):
+    """Change a metadataset"""
+    auth_user = security.revalidate_user(request)
+    db = request.dbsession
+    settings_id = request.matchdict["id"]
+
+    # Only site admins can change site settings
+    if not auth_user.site_admin:
+         raise HTTPForbidden()
+
+    target_settings = resource_by_id(db, ApplicationSettings, settings_id)
+
+    key = request.openapi_validated.body["key"]
+    value_type = request.openapi_validated.body["value_type"]
```

Comment:
Giving it some thought I don't think the value type should be changeable. When accessing the application settings we use hard coded `.str_value` etc to access the values.